src/DateTimePicker.purs

```diff
@@ -162,7 +163,9 @@ render { date, time, targetDate, disabled } =
     , HH.div
       [ css "flex-1" ]
       [ HH.slot _timepicker unit TimePicker.component
-        { disabled
+        { disabled:
+            disabled
+              || (Data.Maybe.isJust interval && Data.Maybe.isNothing date)
```

Comment:
It looks like that behavior was added after all, making disabling the TimePicker unnecessary. Would you mind removing the disabling?

Reply:
If we remove `|| (Data.Maybe.isJust interval && Data.Maybe.isNothing date)`, then I think it is the behavior you are describing.

The dataflow is like
- user selects a Date and DatePicker raise `SelectionChanged _`
  - DatePicker -> DateTimePicker
- DateTimePicker receives and updates selected Date and re-computes `interval` for TimePicker
  - DateTimePicker -> TimePicker
- TimePicker receives new `interval`, if current selection is out of range, set `Nothing` and raise `SelectionChanged Nothing`
  - TimePicker -> DateTimePicker
- DateTimePicker receives and updates selected Time and raise `SelectionChanged Nothing`

Do you prefer without disabling TimePicker? I thought it sort of indicates the desired ordering so user won't need to deal with invalid Time.

Or when selected DateTime is invalid given a DateTime `interval`, instead of only clearing Date and disabling TimePicker, we clear both Date and Time?

Reply:
Didn't see your second comment until hard refresh. Yeah I can do that.